dom/dom.go

```diff
@@ -12,7 +12,7 @@ func init() {
 	SetupXPathCallback()
 	docPool = sync.Pool{}
 	docPool.New = func() interface{} {
-		return Document{}
+		return &Document{}
```

Comment:
Yes, I get it. I fixed it myself elsewhere, but please don't mix things in the same PR

Reply:
Reverted